mypy/fastparse2.py

```diff
@@ -872,68 +891,72 @@ def visit_Num(self, new: ast27.Num) -> Expression:
         elif isinstance(value, complex):
             expr = ComplexExpr(value)
         else:
-            raise RuntimeError('num not implemented for ' + str(type(new.n)))
+            raise RuntimeError('num not implemented for ' + str(type(n.n)))
 
         if is_inverse:
             expr = UnaryExpr('-', expr)
 
-        return expr
+        return self.set_line(expr, n)
 
     # Str(string s)
-    @with_line
-    def visit_Str(self, s: ast27.Str) -> Expression:
+    def visit_Str(self, n: ast27.Str) -> Expression:
         # Hack: assume all string literals in Python 2 stubs are normal
         # strs (i.e. not unicode).  All stubs are parsed with the Python 3
         # parser, which causes unprefixed string literals to be interpreted
         # as unicode instead of bytes.  This hack is generally okay,
         # because mypy considers str literals to be compatible with
         # unicode.
-        if isinstance(s.s, bytes):
-            n = s.s
+        if isinstance(n.s, bytes):
+            s = n.s
```

Comment:
I would use a longer name here instead of `s`, maybe `string`. Also `s` used in above methods for statements.